### PR TITLE
added AdminFlag

### PR DESCRIPTION
**Pull request by muhammad-ali-e**
## What

- Introduced a `ADMIN_ENABLED` flag to control Django admin functionality. Default value set to `False`.
- If `ADMIN_ENABLED` is `True`, Django admin URLs are enabled.

[**UPDATES**]
- Reverted changes that added in existing migrations
- This PR only contains `ADMIN_ENABLED` flag changes

## Why

- Django admin functionality is optional and currently may not be necessary.

## How

- Created a `ADMIN_ENABLED` flag to toggle Django admin functionality.

## Notes on Testing

- Accessing the admin page without enabling it (ADMIN_ENABLED set to False) should result in a 404 error.

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/backend/settings/base.py [hunk 1]
```diff
@@ -143,6 +143,14 @@ def get_required_setting(
 
 DEFAULT_AUTH_USERNAME = os.environ.get("DEFAULT_AUTH_USERNAME", "unstract")
 DEFAULT_AUTH_PASSWORD = os.environ.get("DEFAULT_AUTH_PASSWORD", "unstract")
+SYSTEM_ADMIN_USER = os.environ.get("SYSTEM_ADMIN_USER", "unstract_admin")
+SYSTEM_ADMIN_PASSWORD = os.environ.get(
+    "SYSTEM_ADMIN_PASSWORD", "unstract_admin"
```
Comment (ritwik-g) [inline backend/backend/settings/base.py:148]: @muhammad-ali-e please remove the default value and use `get_required_setting` function.

File: backend/backend/settings/base.py [hunk 2]
```diff
@@ -143,6 +143,14 @@ def get_required_setting(
...
+)
+SYSTEM_ADMIN_EMAIL = os.environ.get("SYSTEM_ADMIN_EMAIL", "admin@abc.com")
```
Comment (ritwik-g) [inline backend/backend/settings/base.py:150]: @muhammad-ali-e please remove the default value and use get_required_setting function.

File: backend/backend/public_urls.py [hunk 3]
```diff
@@ -37,4 +34,13 @@
     # Feature flags
     path(f"{path_prefix}/flags/", include("feature_flag.urls")),
 ]
+if settings.ADMIN_ENABLED is True:
```
Comment (ritwik-g) [inline backend/backend/public_urls.py:37]: @muhammad-ali-e no need for `is True`. In python it is a redundant check.
```suggestion
if settings.ADMIN_ENABLED:
```

File: run-platform.sh [hunk 4]
```diff
@@ -163,15 +163,19 @@ setup_env() {
...
-      # Add default auth credentials for backend.
+      # Add default auth and system admin credentials for backend.
       if [ "$service" == "backend" ]; then
-        echo -e "$blue_text""Adding default auth credentials to $service""$default_text"
+        echo -e "$blue_text""Adding default auth and system admin credentials to $service""$default_text"
         if [[ "$OSTYPE" == "darwin"* ]]; then
           sed -i '' "s/DEFAULT_AUTH_USERNAME.*/DEFAULT_AUTH_USERNAME=\"$DEFAULT_AUTH_KEY\"/" $env_path
           sed -i '' "s/DEFAULT_AUTH_PASSWORD.*/DEFAULT_AUTH_PASSWORD=\"$DEFAULT_AUTH_KEY\"/" $env_path
+          sed -i '' "s/SYSTEM_ADMIN_USERNAME.*/SYSTEM_ADMIN_USERNAME=\"$DEFAULT_AUTH_KEY\"/" $env_path
+          sed -i '' "s/SYSTEM_ADMIN_PASSWORD.*/SYSTEM_ADMIN_PASSWORD=\"$DEFAULT_AUTH_KEY\"/" $env_path
```
Comment (ritwik-g) [inline run-platform.sh:173]: @muhammad-ali-e what will this lines do? I don't think we need to configure this in the scripts since we are not yet planning to enable the admin part.

File: backend/backend/settings/base.py [hunk 5]
```diff
@@ -143,6 +143,14 @@ def get_required_setting(
 
 DEFAULT_AUTH_USERNAME = os.environ.get("DEFAULT_AUTH_USERNAME", "unstract")
 DEFAULT_AUTH_PASSWORD = os.environ.get("DEFAULT_AUTH_PASSWORD", "unstract")
```
Comment (ritwik-g) [inline backend/backend/settings/base.py:145]: @muhammad-ali-e even these in my opinion we should use `get_required_setting` and remove the default values. cc: @hari-kuriakose

**Review by hari-kuriakose — CHANGES_REQUESTED** [review]
@muhammad-ali-e Request to add new migration files with these changes. Please check comments.

File: run-platform.sh (hunk 4 above)
Comment (muhammad-ali-e) [reply run-platform.sh:173]: This values are mandatory for initial migrations.

File: backend/account/migrations/0002_auto_20230718_1040.py [hunk 6]
```diff
@@ -1,30 +1,31 @@
 # mypy: ignore-errors
```
[the remark this answers is not in the document]
Comment (muhammad-ali-e) [reply backend/account/migrations/0002_auto_20230718_1040.py:1]: There is no functionality changes here, These are only cleanup. The variable names are changed due to sonar issues.

File: backend/backend/public_urls.py (hunk 3 above)
Comment (muhammad-ali-e) [reply backend/backend/public_urls.py]: updated

File: backend/backend/settings/base.py (hunk 2 above)
Comment (muhammad-ali-e) [reply backend/backend/settings/base.py]: updated with required changes

File: backend/backend/settings/base.py (hunk 1 above)
Comment (muhammad-ali-e) [reply backend/backend/settings/base.py]: updated with required changes

Comment (muhammad-ali-e) [reply backend/backend/settings/base.py]: updated with required changes

**Review by ritwik-g — APPROVED** [review]

File: backend/account/migrations/0002_auto_20230718_1040.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (hari-kuriakose) [reply backend/account/migrations/0002_auto_20230718_1040.py:1]: @muhammad-ali-e OK. For a fresh setup, this would work theoretically.

However what about an existing setup like Staging or Prod? I don't think Django will pick up changes in already done migrations. That is not recommended either.

[the remark this answers is not in the document]
Comment (muhammad-ali-e) [reply backend/account/migrations/0002_auto_20230718_1040.py:1]: @hari-kuriakose Django will not re-run an migration that already processed. Here My inetntion is to remove those hardcoded values from the migration file  and for the fresh setup they can use the username and password from the ENV.

If we create a new migration file without altering the file those hardocded values still be there in repo.

[the remark this answers is not in the document]
Comment (muhammad-ali-e) [reply backend/account/migrations/0002_auto_20230718_1040.py]: @hari-kuriakose reverted those changes.

[the remark this answers is not in the document]
Comment (hari-kuriakose) [reply backend/account/migrations/0002_auto_20230718_1040.py]: @muhammad-ali-e Sorry for the confusion. Let me rephrase this, for this specific scenario.

- Regarding **fresh installations** - Keep the changes to modify existing migration file, this will remove the hard coded values as you mentioned. Agree here.
- Regarding **updating existing installations** - Are we going to add a new data migration or instead mention in the release notes to do a one-time manual db update to change password?

[the remark this answers is not in the document]
Comment (muhammad-ali-e) [reply backend/account/migrations/0002_auto_20230718_1040.py]: Here, we've opted to implement the changes outlined in the migration files through a separate PR, enhancing our ability to track and maintain a clear history.
cc: @hari-kuriakose

[the remark this answers is not in the document]
Comment (muhammad-ali-e) [reply backend/account/migrations/0002_auto_20230718_1040.py]: Tagging the PR here:  https://github.com/Zipstack/unstract/pull/214

**Review by jaseemjaskp — APPROVED** [review]

**Review by hari-kuriakose — APPROVED** [review]
@muhammad-ali-e LGTM
